pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -187,6 +188,29 @@ func remaining(t time.Time) time.Duration {
 	return t.Sub(time.Now()).Round(time.Second)
 }
 
+func dumpClusterAutoscalerLogs(client runtimeclient.Client, restClient *rest.RESTClient) {
+	pods := corev1.PodList{}
+	caLabels := map[string]string{
+		"app": "cluster-autoscaler",
+	}
+	if err := client.List(context.TODO(), &pods, []runtimeclient.ListOptionFunc{runtimeclient.MatchingLabels(caLabels)}...); err != nil {
+		glog.Errorf("Error querying api for clusterAutoscaler pod object: %v", err)
+		return
+	}
+	// We're only expecting one pod but let's log from all that
+	// are found. If we see more than one that's indicative of
+	// some unexpected problem and we may as well dump its logs.
+	for i, pod := range pods.Items {
+		req := restClient.Get().Namespace(e2e.TestContext.MachineApiNamespace).Resource("pods").Name(pod.Name).SubResource("log")
+		res := req.Do()
+		raw, err := res.Raw()
+		if err != nil {
+			glog.Errorf("Unable to get pod logs: %v", err)
```

Comment:
Should we not `continue` here?

Reply:
We need to handle the case where this can fail. e.g., https://github.com/openshift/cluster-api-actuator-pkg/pull/88/commits/2c1c380483f4d90d324122f50d50b3373995a919#diff-88da310e1da5f7c75bfb19be49e0a89fR207